ddtrace/llmobs/_llmobs.py

```diff
@@ -75,6 +89,164 @@ def disable(cls) -> None:
         cls.enabled = False
         log.debug("%s disabled", cls.__name__)
 
+    def _start_span(
+        self,
+        operation_kind: str,
+        name: Optional[str] = None,
+        session_id: Optional[str] = None,
+        model_name: Optional[str] = None,
+        model_provider: Optional[str] = None,
+    ) -> Span:
+        if name is None:
+            name = operation_kind
+        span = self.tracer.trace(name, resource=operation_kind, span_type=SpanTypes.LLM)
+        span.set_tag_str(SPAN_KIND, operation_kind)
+        if session_id is not None:
+            span.set_tag_str(SESSION_ID, session_id)
+        if model_name is not None:
+            span.set_tag_str(MODEL_NAME, model_name)
+        if model_provider is not None:
+            span.set_tag_str(MODEL_PROVIDER, model_provider)
+        return span
+
+    @classmethod
+    def llm(
+        cls,
+        model_name: str,
+        name: Optional[str] = None,
+        model_provider: Optional[str] = None,
+        session_id: Optional[str] = None,
+    ) -> Span:
+        if cls.enabled is False or cls._instance is None:
+            raise Exception("LLMObs.llm() cannot be used while LLMObs is disabled.")
+        if not model_name:
+            raise ValueError("model_name must be the specified name of the invoked model.")
+        if model_provider is None:
+            model_provider = "custom"
+        return cls._instance._start_span(
+            "llm", name, model_name=model_name, model_provider=model_provider, session_id=session_id
+        )
+
+    @classmethod
+    def tool(cls, name: Optional[str] = None, session_id: Optional[str] = None) -> Span:
```

Comment:
Note to self: need to propagate session_id downwards so that users don't need to set it explicitly for every single span